index.js

```diff
@@ -172,80 +172,80 @@
 
   TypeClass['@@type'] = 'sanctuary-type-classes/TypeClass';
 
-  //  data Location = Constructor | Value
+  //  Prefix :: String
+  var Prefix = 'fantasy-land/';
 
-  //  Constructor :: Location
-  var Constructor = 'Constructor';
-
-  //  Value :: Location
-  var Value = 'Value';
-
-  //  _funcPath :: (Boolean, Array String, a) -> Nullable Function
-  function _funcPath(allowInheritedProps, path, _x) {
-    var x = _x;
-    for (var idx = 0; idx < path.length; idx += 1) {
-      var k = path[idx];
-      if (x == null || !(allowInheritedProps || has(k, x))) return null;
-      x = x[k];
+  //  funcPath :: (Array String, a) -> Nullable Function
+  function funcPath(path, value) {
+    var x = value;
+    for (var i = 0; i < path.length; i += 1) {
+      x = x && x[path[i]];
+      if (!x) return null;
     }
     return typeof x === 'function' ? x : null;
```

Comment:
This is one of the few changed lines, but after the simplification of the algorithm, the `null`-branch was never taken. I added this `Bad()`-structure to the tests in order to create the input necessary to reach this branch. If you think the `Bad()` input should actually just lead to a TypeError then we can remove the ternary and the new test cases.